### PR TITLE
Replaces priority

### PR DESCRIPTION
**Pull request by xnox**
This makes things awesome for python.

### Functional Changes

- [ ] This change can build all of Wolfi without errors (describe results in notes)

Notes:

### SCA Changes

- [ ] Examining several representative APKs show no regression / the desired effect (details in notes)

Notes:

### Linter

- [ ] The new check is clean across Wolfi
- [ ] The new check is opt-in or a warning

Notes:

### CODE REVIEW

File: pkg/config/schema.json [hunk 1]
```diff
@@ -179,8 +179,12 @@
           "description": "Optional: List of replace objectives"
         },
         "provider-priority": {
-          "type": "integer",
-          "description": "Optional: An integer compared against other equal package provides used to\ndetermine priority"
+          "type": "string",
+          "description": "Optional: An integer compared against other equal package provides used to\ndetermine priority of provides"
```
[the remark this answers is not in the document]
Comment (xnox) [reply pkg/config/schema.json:183]: I'm not sure how to validate this, as with the above changes nothing checks that priorities convert to integer-looking-string. Horum.

**Comment by kaniini** [comment]
with my apk-tools maintainer hat on, this looks fine

**Comment by xnox** [comment]
@kaniini thank you for your comments. "replaces_priority" is only available in the installed database as `q`, but it is not available in the APKINDEX, meaning looking at APKINDEX alone one cannot know how to order things ahead of time. Would that be just an oversight? as "provider_priority" is available under `k` in both installled database and apkindex.

Or is it optional and it wouldn't hurt anything to add it in the APKINDEX?

**Review by xnox — COMMENTED** [review]
we need this.

**Comment by xnox** [comment]
> Some doc on `range` would be nice. I can't seem to find anything in doc/ about `range` at all. But I understand you didn't add this.

There is https://github.com/chainguard-dev/melange/blob/main/NEWS.md#major-changes-from-010-to-020 and also schema (but it just says that range is an acceptable key, without explaining which arguments it supports)

> 
> It looks like it would be pretty easy to add test coverage to this in `pkg/config/config_test.go` . Could you add some coverage?

Will check.

**Comment by xnox** [comment]
Tests added. Docs are in academy and in the repo.

Will see where best to improve stuff.

**Comment by smoser** [comment]
Just a comment here for myself on our intended use case (per @xnox)
The goal is for us to create py3.10-foo, py3.11-foo, py3.12-foo (each supported python version, as we already do in some cases).  We want the py3.12-foo (the latest) to be "default", such that if something depends on `py3-foo` it will get the highest priority version without us having to encode that specifically in py3-foo.yaml.

xnox wrote:

> all three py3.x-foo packages can provide py3-foo and highest one gets automatically installed without encoding which one is default in the top level py3-foo package as explicit dep.
> so "provider-priority:" variable substitution will be used
> and "replaces-priority" will be supported but in-so-far unused

**Review by smoser — APPROVED** [review]
thanks for adding a test!

**Pull request merged**